File: TemplateWebExtension/src/Services/interfaces/IMyTestService.ts
import { serviceName } from "@docsvision/web/core/services";
import {IMyTestRequestModel} from "../../Models/IMyTestRequestModel";
import {IMyTestModel} from "../../Models/IMyTestModel";
import { IMyTestMemberRequestModel } from "../../Models/IMyTestMemberRequestModel";
import { IMyTestDateRequsetModel } from "../../Models/IMyTestDateRequestModel";
import { IMyTicketRequestModel } from "../../Models/IMyTicketRequestModel";

export interface IMyTestService{
    GetName(model: IMyTestRequestModel): Promise<IMyTestModel>;
    GetMemberSentData(model: IMyTestMemberRequestModel): Promise<IMyTestModel>;
    ChangeMoneyData(model: IMyTestMemberRequestModel): Promise<IMyTestModel>;
    ChangeDayCount(model: IMyTestDateRequsetModel): Promise<IMyTestModel>;
    GetTripData(model: IMyTestRequestModel): Promise<IMyTestModel>;
    GetTestData(model: IMyTestRequestModel): Promise<IMyTestModel>;
    GetTicketsData(model: IMyTicketRequestModel): Promise<IMyTestModel>;
}

export type $MyTestService = { activityPlanService: IMyTestService };
export const $MyTestService = serviceName<$MyTestService, IMyTestService>(x => x.activityPlanService)

File: MyTestServerExtension/Services/IMyTestService.cs
using DocsVision.Platform.WebClient;
using MyTestServerExtension.Model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MyTestServerExtension.Services
{
    public interface IMyTestService{
        MyTestModel GetName(SessionContext sessionContext,Guid cardId);
        MyTestModel GetMemberSentData(SessionContext sessionContext, Guid cardId, Guid userId);
        MyTestModel ChangeMoneyData(SessionContext sessionContext, Guid cardId, Guid cityId);

        MyTestModel ChangeDayCount(SessionContext sessionContext, Guid cardId, string dateFrom, string dateTo);

        public void InitMyCard(SessionContext sessionContext, Guid cardId);

        public MyTestModel GetTripData(SessionContext sessionContext, Guid cardId);
        public MyTestModel GetTestData(SessionContext sessionContext, Guid cardId);

        public MyTestModel GetTicketData(SessionContext sessionContext, Guid cardId, string dateFrom, string dateTo, Guid cityRef);
    }
}
